nomad/plan_apply.go

```diff
@@ -88,18 +100,29 @@ func (p *planner) planApply() {
 			return
 		}
 
-		// Check if out last plan has completed
+		// If last plan has completed get a new snapshot
 		select {
-		case <-waitCh:
-			waitCh = nil
+		case idx := <-planIndexCh:
+			// Previous plan committed. Discard snapshot and ensure
+			// future snapshots include this plan.
+			prevPlanResultIndex = max(prevPlanResultIndex, idx)
```

Comment:
Curious if this defensiveness or a possible case?  Can we get a case where the index of the just committed plan is older than that of the previous plan?

Also, how should we handle if `asyncPlanWait` fails and the channel was closed without a value?  I assume the code is fine (as `idx` being 0) , but the comment of plan being committed might not be accurate.

Reply:
Great questions!

> Can we get a case where the index of the just committed plan is older than that of the previous plan?

No.

> Also, how should we handle if asyncPlanWait fails and the channel was closed without a value? I assume the code is fine (as idx being 0) , but the comment of plan being committed might not be accurate.

Yes, the intent of `max(prev, cur)` here is to drop 0 value from failed plan applications. I'll update the comment to reflect that.